mne/io/meas_info.py

```diff
@@ -815,37 +815,57 @@ def ch_names(self):
         return self['ch_names']
 
     def _repr_html_(self, caption=None):
+        """Summarize info for HTML representation."""
         if isinstance(caption, str):
             html = f'<h4>{caption}</h4>'
         else:
             html = ''
-        n_eeg = len(pick_types(self, meg=False, eeg=True))
-        n_grad = len(pick_types(self, meg='grad'))
-        n_mag = len(pick_types(self, meg='mag'))
-        n_fnirs = len(pick_types(self, meg=False, eeg=False, fnirs=True))
-        pick_eog = pick_types(self, meg=False, eog=True)
-        if len(pick_eog) > 0:
-            eog = ', '.join(np.array(self['ch_names'])[pick_eog])
-        else:
+
+        # good channels
+        channels = {}
+        ch_types = [channel_type(self, idx) for idx in range(len(self['chs']))]
+        ch_counts = Counter(ch_types)
+        for ch_type, count in ch_counts.items():
+            if ch_type == 'meg':
+                channels['mag'] = len(pick_types(self, meg='mag'))
+                channels['grad'] = len(pick_types(self, meg='grad'))
+            elif ch_type == 'eog':
+                pick_eog = pick_types(self, eog=True)
+                eog = ', '.join(
+                    np.array(self['ch_names'])[pick_eog])
+            elif ch_type == 'ecg':
+                pick_ecg = pick_types(self, ecg=True)
+                ecg = ', '.join(
+                    np.array(self['ch_names'])[pick_ecg])
+            channels[ch_type] = count
+
+        good_channels = ', '.join(
+            [f'{v} {k.upper()}' for k, v in channels.items()])
```

Comment:
On `main` I get:
![Screenshot from 2021-09-15 13-58-04](https://user-images.githubusercontent.com/2365790/133485311-803d0f1a-43fe-41be-8d7f-fa74be92004c.png)

On this PR it's a little bit less nice:

![Screenshot from 2021-09-15 13-58-27](https://user-images.githubusercontent.com/2365790/133485341-94835467-c86a-4ff2-b3ff-4fb68b5def93.png)

I prefer the longer / complete names, so some variant of this:
```suggestion
        titles = 
            [f'{v} {titles.get(k, k)}' for k, v in channels.items()])
```
I'll push a commit, can you see if you agree @vagechirkov ?

Reply:
I think it looks better with full names.